lib/invisible_captcha/controller_ext.rb

```diff
@@ -70,11 +99,28 @@ def timestamp_spam?(options = {})
 
       return false
     end
-
-    def honeypot_spam?(options = {})
+    
+    def ip_spam?(options ={})
       honeypot = options[:honeypot]
       scope    = options[:scope] || controller_name.singularize
 
+      if InvisibleCaptcha.ip_enabled
+        if params[:spinner] == spinner_value || (params[scope] && params[scope][:spinner] == spinner_value) 
```

Comment:
sorry for insisting on this, but all these things can be simplified if we found a way to make all this work under the hood (as we do for example with the time sensitive check, final users only need to enable/disable)

Reply:
I completely agree with everything you are saying.  The main difference between the time sensitive check and the spinner is that we need to set values from the controller and make them available to the views.  However, some need to set the values and some actions need to detect them.  If the helper_method issue was solved, we would not need the before_action and I think the API would be left alone.